server/controllers/v1/user.ts

```diff
@@ -0,0 +1,92 @@
+import argon2 from "argon2";
+import jwt from "jsonwebtoken";
+import { Request, Response } from "express";
+
+import db from "../../models/index";
+import { User } from "../../models/user";
+
+export default class UserController {
+  static secret = process.env.SECRET as string;
+  static UserModel: typeof User = db.User;
+
+  static updateLastLogin = async (user: User) => {
+    user.lastLogin = new Date();
+    await user.save();
+  };
+
+  static signIn = async (req: Request, res: Response) => {
+    const { userName, password } = req.body;
+
+    try {
+      const oldUser = await UserController.UserModel.findOne({
+        where: {
+          userName: userName,
+        },
+      });
+      if (!oldUser)
+        return res.status(404).json({ message: "User doesn't exist" });
+
+      const isPasswordCorrect = await argon2.verify(
+        oldUser.password,
+        password,
+        {
+          type: argon2.argon2id,
+        }
+      );
+      if (!isPasswordCorrect)
+        return res.status(400).json({ message: "Invalid credentials" });
+
+      const token = jwt.sign(
+        { username: oldUser.userName, id: oldUser.id },
+        UserController.secret,
+        { expiresIn: "1h" }
+      );
+
+      await UserController.updateLastLogin(oldUser);
+
+      res.status(200).json({ result: oldUser, token });
+    } catch (error) {
+      res.status(500).json({ message: "Something went wrong" });
+    }
+  };
+
+  static createUser = async (req: Request, res: Response) => {
+    const { email, userName, password, firstName, lastName } = req.body;
+
+    try {
+      const oldUser = await UserController.UserModel.findOne({
+        where: {
+          userName: userName,
+        },
+      });
+      if (oldUser)
```

Comment:
same thing, I think you should use `catch` to handle db errors

Reply:
The catch statement can be found below at line 86